src/serum.rs

```diff
@@ -319,6 +151,239 @@ impl<'a> Market<'a> {
             open_orders_accounts: Vec::new(),
         })
     }
+
+    /// Creates a new order and pushes it to the sandbox -
+    /// will fail if the transaction does not go through.
+    /// It is important to note that matching occurs at this state
+    /// inside of Serum itself in V3, however, in earlier versions,
+    /// this does not occur until requests are popped off of the request queue.
+    pub fn new_order(
+        &self,
+        payer: &Actor<'a>,
+        participant: &Participant<'a>,
+        side: Side,
+        limit_price: NonZeroU64,
+        order_type: OrderType,
+        max_base_qty: NonZeroU64,
+        client_order_id: u64,
+        self_trade_behavior: SelfTradeBehavior,
+        limit: u16,
+        max_native_quote_including_fees: NonZeroU64,
+        srm_account_referral: Option<&Pubkey>,
+    ) -> Result<()> {
+        let new_order_instruction = serum_dex::instruction::new_order(
+            self.market.pubkey(),
+            participant.open_orders().pubkey(),
+            self.request_queue.pubkey(),
+            self.event_queue.pubkey(),
+            self.bids.pubkey(),
+            self.asks.pubkey(),
+            payer.pubkey(),
+            participant.account().pubkey(),
+            self.base_vault.account().pubkey(),
+            self.quote_vault.account().pubkey(),
+            &spl_token::ID,
+            &solana_program::sysvar::rent::ID,
+            srm_account_referral,
+            self.serum,
+            side,
+            limit_price,
+            max_base_qty,
+            order_type,
+            client_order_id,
+            self_trade_behavior,
+            limit,
+            max_native_quote_including_fees,
+        )?;
+
+        self.sandbox.send_signed_transaction_with_payers(
+            &[new_order_instruction],
+            Some(participant.account.pubkey()),
+            vec![participant.account.keypair()],
+        )
+    }
+
+    /// Spin up consume_events_loop on another thread and kill it after
```

Comment:
Got it. I guess as an alternative, it would be nice in the future to add some helper code to "await the consumption of your event." In pseudocode, something like:

```
event = ....?
let loop = foo.consume_events_loop(...);
event.block_until_consumed();
loop.kill();
```

Reply:
Created https://github.com/foonetic/solarium/issues/15